### PR TITLE
feat: delay closing wss connection when app is paused

### PR DESCRIPTION
**Pull request by jakubuid**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,7 +1,7 @@
 android.useAndroidX=true
 
 GROUP=com.tinder.scarlet
-VERSION_NAME=0.1.13-SNAPSHOT
+VERSION_NAME=0.1.14-SNAPSHOT
```
Comment (Talhaali00) [inline gradle.properties:4]: just a heads up, this won't be used to generate the artifact version. The version will be determined by the release name

File: scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt [hunk 2]
```diff
@@ -10,22 +10,53 @@ import android.os.Bundle
...
-        )
+        var isResumed: Boolean = false
+        var job: Job? = null
 
-        override fun onActivityResumed(activity: Activity) = lifecycleRegistry.onNext(Lifecycle.State.Started)
+        override fun onActivityPaused(activity: Activity) {
+            isResumed = false
+
+            job = scope.launch {
+                supervisorScope {
```
Comment (Talhaali00) [inline scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt]: we don't need an additional `SupervisorScope`.

File: scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt [hunk 3]
```diff
@@ -10,22 +10,53 @@ import android.os.Bundle
...
+                    delay(30000)
```
Comment (Talhaali00) [inline scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt]: We should use a variable or a function like `TimeUnit.SECONDS.toMillis(30)`

File: scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt [hunk 4]
```diff
@@ -10,22 +10,53 @@ import android.os.Bundle
...
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
 
 internal class ApplicationResumedLifecycle(
     application: Application,
     private val lifecycleRegistry: LifecycleRegistry
 ) : Lifecycle by lifecycleRegistry {
+    private val job = SupervisorJob()
+    private var scope = CoroutineScope(job + Dispatchers.IO)
```
Comment (Talhaali00) [inline scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt]: Let's use Dispatchers.Default here

**Review by Talhaali00 — APPROVED** [review]

File: gradle.properties (hunk 1 above)
Comment (jakubuid) [reply gradle.properties:4]: Ah okay, good point
So this is irrelevant

File: scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt (hunk 2 above)
Comment (jakubuid) [reply scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt]: That's true 👌

File: scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt (hunk 3 above)
Comment (jakubuid) [reply scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt]: Yeah, will change that

File: scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt (hunk 4 above)
Comment (jakubuid) [reply scarlet-lifecycle-android/src/main/java/com/tinder/scarlet/lifecycle/android/ApplicationResumedLifecycle.kt]: Makes sense

**Pull request merged**